zone/client_process.cpp

```diff
@@ -2352,3 +2354,11 @@ void Client::SendGuildLFGuildStatus()
 	worldserver.SendPacket(pack);
 	safe_delete(pack);
 }
+
+bool Client::CheckWaterAutoFireLoS(Mob* los_attacker, Mob* los_target) // checks if both attacker and target are both in or out of the water
+{
+	if (!RuleB(Combat, WaterMatchRequiredForAutoFireLoS)) { // if rule is set to false, bypass check
+		return true;
+	}
+	return zone->watermap->InLiquid(los_attacker->GetPosition()) == zone->watermap->InLiquid(los_target->GetPosition());
```

Comment:
Probably need to check `zone->watermap` isn't nullptr for the crash mentioned in comments. Both Mob* arguments should be valid when this is called since `GetTarget()` is checked already in the caller

Reply:
This function should probably take a single argument, otherwise, what's the point of this being a member function?

Reply:
@mackal How would you recommend doing it? New to this just wanted to PR this if anyone wants as it's a feature I feel is missing for a "classic" experience.

Reply:
```c++
bool Client::CheckWaterAutoFireLoS(Mob* los_target) // checks if both attacker and target are both in or out of the water
{
	if (!RuleB(Combat, WaterMatchRequiredForAutoFireLoS) || zone->watermap == nullptr || los_target == nullptr) { // if rule is set to false, bypass check
		return true;
	}
	return zone->watermap->InLiquid(GetPosition()) == zone->watermap->InLiquid(los_target->GetPosition());
}
```